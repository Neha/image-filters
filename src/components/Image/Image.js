import React, { useState } from 'react';
import Filters from '../Filters/Filters';
import PropTypes from 'prop-types';

const Image = props => {
  const [classes, setClasses] = useState([]);
  const imgName = props.imgName;

  function setFilter(className) {
    let _classes = Object.assign([], classes);
    if (className !== "reset") {
      if (!_classes.includes(className)) {
        _classes.push(className);
      } else {
        _classes.pop(className);
      }

      setClasses(_classes);
    } else {
      _classes = [];
      setClasses(_classes);
    }
  }

  return (
    <section>
      <div>
        <img src={imgName} className={classes.join(" ")} />
      </div>
      {imgName && <Filters onClick={setFilter} tabindex="1" />}
    </section>
  );
};

Image.propTypes = {
  imgName: PropTypes.string
};

export default Image;
