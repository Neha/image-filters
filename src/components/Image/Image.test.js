import React from "react";
import ReactDOM from "react-dom";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
configure({ adapter: new Adapter() });

import { shallow } from "enzyme";
import Image from './Image';

describe('Image', () => {
    it("Image renders without crashing", () => {
        const wrapper = shallow(<Image />);
        expect(wrapper.exists()).toBe(true);
      });
    
      it("Image Snapshot testing", () => {
        const tree = renderer.create(<Image />).toJSON();
        expect(tree).toMatchSnapshot();
      });
})