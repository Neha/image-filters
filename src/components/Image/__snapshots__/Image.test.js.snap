// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Image Snapshot testing 1`] = `
<section>
  <div>
    <img
      className=""
    />
  </div>
</section>
`;
